### DISCUSSION TITLE
DefinePlugin: typeof usage?

### DISCUSSION THREAD

**Discussion by pcafstockf** [webpack/webpack#14107] [Q&A]
Can anyone describe how the following config (from the DefinePlugin documentation) would be used?
```
new webpack.DefinePlugin({
  'typeof window': JSON.stringify('object')
});
```
I've searched docs, examples, test cases, google, and even skimmed the plugin code.  But I just don't understand the purpose or usage of this `typeof` prefix.

**Comment by alexander-akait** [comment]
You can use `typeof something` in your code

**Reply by pcafstockf** [reply]
Well yes, the documentation does suggest that :smile:
Would you be so kind as to show an example usage?

Something like the examples the documentation provides, **but** with `typeof`:
```
new webpack.DefinePlugin({
  PRODUCTION: JSON.stringify(true),
  'typeof window': JSON.stringify('object'),
});
```
Input
```
if (!PRODUCTION) {
  console.log('Debug info');
}
```
Output:
```
if (!true) {
  console.log('Debug info');
}

```
